### PR TITLE
Clarify use of enum in union types

### LINKED ISSUE #712: Clarify usage of `enum` in a union type
Any type declaration may have an `enum` facet. In the context of a [union type](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#union-type), this means that each value of the `enum` should match at least one of the super type declaration. We should make that clear.

### Proposed clarification
- Add that `if the facet "enum" is declared in a union type, each value of that "enum" MUST match at least one of its super types`

### PR DESCRIPTION
**Pull request by jstoiko**
Closes #712

### CODE REVIEW

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
 types:
    HomeAnimal: [ HasHome | IsOnFarm ,  Dog | Cat | Parrot ]
 ```
+If a union type contains a facet with an `enum`, every value of that `enum` MUST meet all restrictions associated with at least one of the super types. Here is an example:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1172]:
```suggestion
If a union type contains a facet with an `enum`, every value of that `enum` must meet all the restrictions associated with at least one of the super types:
```

File: versions/raml-10/raml-10.md [hunk 2]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+
+This is valid:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1174]:
```suggestion
The following example illustrates a valid expression:
```

File: versions/raml-10/raml-10.md [hunk 3]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+```
+type: number | boolean
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1176]:
```suggestion
`type: number | boolean`
```

File: versions/raml-10/raml-10.md [hunk 4]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+enum: [1, true, 2]
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1177]:
```suggestion
`enum: [1, true, 2]`
```

File: versions/raml-10/raml-10.md [hunk 5]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+```
+
+This is invalid:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1180]:
```suggestion
The following example illustrates an invalid expression:
```

File: versions/raml-10/raml-10.md [hunk 6]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+```
+type: number | boolean
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1182]:
```suggestion
`type: number | boolean`
```

File: versions/raml-10/raml-10.md [hunk 7]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+enum: [1, true, 2, "hello"]
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1183]:
```suggestion
`enum: [1, true, 2, "hello"]`
```

File: versions/raml-10/raml-10.md [hunk 8]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+```
+
+Recall that types in this case can be built-in data types, such as number or boolean, but also custom user-defined types such as unions or complex types with multiple properties. Imagine a more complex example:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1186]:
```suggestion
Note that types, in this case, can be built-in data types, such as numbers or boolean, or can be custom user-defined types, such as unions or complex types with multiple properties. Imagine a more complex example:
```

File: versions/raml-10/raml-10.md [hunk 9]
```diff
@@ -1169,6 +1169,48 @@ If you are extending from two union types a processor MUST perform validations f
...
+
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1187]: Add the whole example in a code tag, like this:
[source,text,linenums]
----
Add the full example here.
----

**Review by priyaayyar — CHANGES_REQUESTED** [review]

File: versions/raml-10/raml-10.md (hunk 6 above)
Comment (jstoiko) [reply versions/raml-10/raml-10.md:1182]: the 3 backticks above are meant to format the whole block below it as if each line were surrounded with single backticks so these are not needed.

File: versions/raml-10/raml-10.md (hunk 7 above)
Comment (jstoiko) [reply versions/raml-10/raml-10.md:1183]: ditto

File: versions/raml-10/raml-10.md (hunk 9 above)
Comment (jstoiko) [reply versions/raml-10/raml-10.md:1187]: Not sure to understand. Is the example below not that?

Comment (dukesphere) [reply versions/raml-10/raml-10.md:1187]: There was a misunderstanding. The [source, text, linenums] is not needed in Markdown here. That relates to a convention of our Asciidoc format.

**Review by dukesphere — APPROVED** [review]
I think it's fine as-is.

**Review by priyaayyar — APPROVED** [review]

**Pull request merged**